pkgs/swiftgen/lib/src/ast/compund_declaration.dart

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'declaration.dart';
+import 'referred_type.dart';
+
+class CompoundDeclaration extends Declaration {
```

Comment:
Can you add a short comment to each of these classes and their fields saying what Swift language concept it represents?

Reply:
I will, but first I added AST models for class, struct, and protocol. I'm using `abstract interface`s to declare common fields. I was extending abstract classes initially, but I felt like interfaces are more flexible.

I'm not sure if this approach has any drawbacks, so can you please check things out first in case there're any drawbacks?

Reply:
The class modifiers seem fine to me. I don't use them very often tbh (other than `abstract`). Is there a reason `ObjcAnnotatable` is an `interface` but not `abstract`?

Reply:
I guess the main drawback is it's a bit verbose. Since the fields are abstract, you're having to redeclare them in all the child classes, with `@override`, so that's 3 lines per field instead of 1.

In the meeting you said you wanted to do this to avoid writing a constructor. If that's the only reason, then I'm pretty sure it would be less code overall to just write the constructor and avoid redeclaring the fields in the child classes.